From: Danny Kukawka <danny.kukawka@web.de>
Date: Mon, 25 Aug 2008 10:58:43 +0000 (+0200)
Subject: added check for ConsoleKit >= v0.3.1 to configure
X-Git-Url: http://gitweb.freedesktop.org/?p=hal.git;a=commitdiff;h=aa673602a42b3fab01b14dfd2934e9f15e4c6428

added check for ConsoleKit >= v0.3.1 to configure

Added check for ConsoleKit >= v0.3.1 to configure to be able
to differ between the versions due to API breakage.
---

--- a/configure.in
+++ b/configure.in
@@ -486,6 +486,20 @@ if test "x$enable_console_kit" != "xno";
    AM_CONDITIONAL(HAVE_CONKIT, true)
    AC_DEFINE(HAVE_CONKIT, [], [Set if we use ConsoleKit])
    msg_conkit=yes
+   # yes this is ugly, but there is no other way to get the version of CK 
+   AC_MSG_CHECKING([if ConsoleKit version 0.3.0 or newer])
+   if $PKG_CONFIG --atleast-version=0.3.0 ck-connector; then
+     AC_MSG_RESULT([yes])
+     AC_DEFINE(HAVE_CK_0_3, 1, [Define to 1 if ConsoleKit is v0.3.0 or newer])
+   else 
+     if $PKG_CONFIG --max-version=0.2.10 ck-connector; then
+       AC_MSG_RESULT([no])
+     else
+       #assume we have the latest version
+       AC_MSG_WARN([Couldn't detect ConsoleKit version, install the devel package, assume for now you use >= 0.3.0])
+       AC_DEFINE(HAVE_CK_0_3, 1, [Define to 1 if ConsoleKit is v0.3.0 or newer])
+     fi
+   fi
 fi
 
 AC_PATH_PROG(GPERF, [gperf], [no])
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Mon, 25 Aug 2008 11:02:19 +0000 (+0200)
Subject: adapt new CK library version (>= v0.3.1) to HAL
X-Git-Url: http://gitweb.freedesktop.org/?p=hal.git;a=commitdiff;h=32022662804d43434a3e7ff168d314385867eb76

adapt new CK library version (>= v0.3.1) to HAL

Adapted new CK library version (>= v0.3.1) to HAL. Adopted
patch from Ben Gamari <bgamari@gmail.com> to work with old
and new ConsoleKit versions.
---

--- a/hald/ck-tracker.c
+++ b/hald/ck-tracker.c
@@ -256,7 +256,11 @@ ck_session_get_info (CKTracker *tracker,
 		goto error;
 	}
 	if (!dbus_message_get_args (reply, NULL,
+#ifdef HAVE_CK_0_3
+				    DBUS_TYPE_UINT32, &(session->user),
+#else
 				    DBUS_TYPE_INT32, &(session->user),
+#endif
 				    DBUS_TYPE_INVALID)) {
 		HAL_ERROR (("Invalid GetUnixUser reply from CK"));
 		goto error;
@@ -531,7 +535,11 @@ ck_tracker_process_system_bus_message (C
 		seat_objpath = dbus_message_get_path (message);
 
 		if (!dbus_message_get_args (message, NULL,
+#ifdef HAVE_CK_0_3
+					    DBUS_TYPE_OBJECT_PATH, &seat_objpath,
+#else
 					    DBUS_TYPE_STRING, &seat_objpath,
+#endif
 					    DBUS_TYPE_INVALID)) {
 			HAL_ERROR (("Invalid SeatAdded signal from CK"));
 			goto out;
@@ -558,7 +566,11 @@ ck_tracker_process_system_bus_message (C
 		seat_objpath = dbus_message_get_path (message);
 
 		if (!dbus_message_get_args (message, NULL,
+#ifdef HAVE_CK_0_3
+					    DBUS_TYPE_OBJECT_PATH, &seat_objpath,
+#else
 					    DBUS_TYPE_STRING, &seat_objpath,
+#endif
 					    DBUS_TYPE_INVALID)) {
 			HAL_ERROR (("Invalid SeatRemoved signal from CK"));
 			goto out;
@@ -588,7 +600,11 @@ ck_tracker_process_system_bus_message (C
 		seat_objpath = dbus_message_get_path (message);
 
 		if (!dbus_message_get_args (message, NULL,
+#ifdef HAVE_CK_0_3
+					    DBUS_TYPE_OBJECT_PATH, &session_objpath,
+#else
 					    DBUS_TYPE_STRING, &session_objpath,
+#endif
 					    DBUS_TYPE_INVALID)) {
 			HAL_ERROR (("Invalid SessionAdded signal from CK"));
 			goto out;
@@ -624,7 +640,11 @@ ck_tracker_process_system_bus_message (C
 		seat_objpath = dbus_message_get_path (message);
 
 		if (!dbus_message_get_args (message, NULL,
+#ifdef HAVE_CK_0_3
+					    DBUS_TYPE_OBJECT_PATH, &session_objpath,
+#else
 					    DBUS_TYPE_STRING, &session_objpath,
+#endif
 					    DBUS_TYPE_INVALID)) {
 			HAL_ERROR (("Invalid SessionRemoved signal from CK"));
 			goto out;
